### PR TITLE
Migrate to v1.6.2

### LINKED ISSUE #31: Update TuxGuitar flatpak to 1.6.0?
The historical dev repository for TuxGuitar seems to be orphaned: refer to [this post](https://sourceforge.net/p/tuxguitar/discussion/522984/thread/0a8c900ddd/). Last app update is more than 1 year old, and this year the TuxGuitar website was discontinued. Even the donation link is no more available.
After [trying to contact](https://sourceforge.net/p/tuxguitar/discussion/522984/thread/02743a8637/) historical developer, a [fork](https://github.com/helge17/tuxguitar) has emerged a few months ago with the objective to continue TuxGuitar maintenance and development.
A new version [1.6.0](https://github.com/helge17/tuxguitar/releases/tag/1.6.0) has been recently released. Could it be possible to update the TuxGuitar flatpak?

I don't know flatpak at all, or I would probably have sent a pull request to suggest this update. However I contributed to the [rpm packaging](https://build.opensuse.org/package/show/openSUSE:Factory/tuxguitar) of version 1.6.0 for openSUSE, so I might be able to provide some help if needed for this update (e.g. patches for tuxguitar.sh or .desktop, etc)

best regards

### PR DESCRIPTION
**Pull request by mdouchement**
Fix #31 #30

### CODE REVIEW

File: ar.com.tuxguitar.TuxGuitar.yml [hunk 1]
```diff
@@ -19,44 +21,42 @@ finish-args:
   - --env=PATH=/usr/bin:/app/bin:/app/jre/bin
   - --env=JAVA_HOME=/app/jre
   - --env=JAVA=/app/jre/bin/java
-  - --env=VST_PATH=/app/extensions/Plugins/vst
```
Comment (hfiguiere) [inline ar.com.tuxguitar.TuxGuitar.yml:22]: why removing this?

Comment (hfiguiere) [reply ar.com.tuxguitar.TuxGuitar.yml:22]: never found the issue. would have been nice to mention it in the PR.

File: tuxguitar.sh [hunk 2]
```diff
@@ -1,31 +1,18 @@
...
+CLASSPATH=${CLASSPATH}:${TG_DIR}/dist/
 ##MAINCLASS
 MAINCLASS=org.herac.tuxguitar.app.TGMainSingleton
 ##JVM ARGUMENTS
 VM_ARGS="-Xmx512m"
-
+##EXPORT VARS
 export CLASSPATH
 export LD_LIBRARY_PATH
-export LV2_PATH=${LV2_PATH:-$HOME/.lv2:/app/extensions/Plugins/lv2:/app/lib/lv2}
```
Comment (hfiguiere) [inline tuxguitar.sh:29]: however why removing this?

File: ar.com.tuxguitar.TuxGuitar.yml [hunk 3]
```diff
@@ -3,11 +3,13 @@ runtime: org.freedesktop.Platform
...
 sdk: org.freedesktop.Sdk
 sdk-extensions:
-  - org.freedesktop.Sdk.Extension.openjdk11
+  - org.freedesktop.Sdk.Extension.openjdk17
 command: tuxguitar
 finish-args:
   - --share=ipc
   - --socket=x11
+  # Access to wayland
+  - --socket=wayland
```
Comment (hfiguiere) [inline ar.com.tuxguitar.TuxGuitar.yml:12]: with wayland socket it shuld be `fallback-x11`

File: ar.com.tuxguitar.TuxGuitar.yml [hunk 4]
```diff
@@ -19,44 +21,42 @@ finish-args:
...
   - name: fluidsynth
     buildsystem: cmake-ninja
     config-opts:
       - -DLIB_SUFFIX=
       - -DCMAKE_BUILD_TYPE=Release
     sources:
       - type: archive
-        url: https://github.com/FluidSynth/fluidsynth/archive/refs/tags/v2.1.9.tar.gz
-        sha256: 365642cc64bafe0491149ad643ef7327877f99412d5abb93f1fa54e252028484
+        url: https://github.com/FluidSynth/fluidsynth/archive/refs/tags/v2.3.5.tar.gz
```
Comment (hfiguiere) [inline ar.com.tuxguitar.TuxGuitar.yml:51]: you should use the one in shared-modules, as indicated by the comment you deleted.

File: ar.com.tuxguitar.TuxGuitar.yml [hunk 5]
```diff
@@ -19,44 +21,42 @@ finish-args:
...
+        sha256: f89e8e983ecfb4a5b4f5d8c2b9157ed18d15ed2e36246fa782f18abaea550e0d
 
   - name: tuxguitar
     buildsystem: simple
     build-commands:
       - mkdir -p /app/bin
-      - tar -xf tuxguitar-*-linux-x86*.tar.gz --strip 1 --exclude=tuxguitar.sh -C /app/
-      - install -Dm644 /app/share/skins/Oxygen/icon-64x64.png /app/share/icons/hicolor/64x64/apps/ar.com.tuxguitar.TuxGuitar.png
+      - tar -xf tuxguitar-*-linux-swt-x86_64.tar.gz --strip 1 --exclude=tuxguitar.sh -C /app/
+      - install -Dm644 /app/share/skins/Oxygen/icon.png /app/share/icons/hicolor/64x64/apps/ar.com.tuxguitar.TuxGuitar.png
```
Comment (hfiguiere) [inline ar.com.tuxguitar.TuxGuitar.yml:59]: that icon is not 64x64.

File: tuxguitar.sh (hunk 2 above)
Comment (mdouchement) [reply tuxguitar.sh:29]: It is due to a copy/paste from https://github.com/helge17/tuxguitar/blob/1.6.2/desktop/build-scripts/common-resources/common-linux/tuxguitar.sh
I'll restore this line

File: ar.com.tuxguitar.TuxGuitar.yml (hunk 5 above)
Comment (mdouchement) [reply ar.com.tuxguitar.TuxGuitar.yml:59]: Yes, I'll fix it.

File: ar.com.tuxguitar.TuxGuitar.yml (hunk 4 above)
Comment (mdouchement) [reply ar.com.tuxguitar.TuxGuitar.yml:51]: Oh you are right

File: ar.com.tuxguitar.TuxGuitar.yml [hunk 6]
```diff
@@ -19,64 +22,53 @@ finish-args:
...
-      - install -d /app/extensions/Plugins
+      - mkdir -p "${FLATPAK_DEST}/bin"
+      - tar -xf tuxguitar-*-linux-swt-x86_64.tar.gz --strip 1 --exclude=tuxguitar.sh -C "${FLATPAK_DEST}/"
+      - install -Dm0644 "${FLATPAK_ID}.png" -t "${FLATPAK_DEST}/share/icons/hicolor/256x256/apps"
+      - install -Dm0644 "${FLATPAK_ID}.metainfo.xml" -t "${FLATPAK_DEST}/share/metainfo/"
+      - install -Dm0644 "${FLATPAK_DEST}/share/skins/Oxygen/icon.png" "${FLATPAK_DEST}/share/icons/hicolor/96x96/apps/${FLATPAK_ID}.png"
+      - install -D tuxguitar.sh "${FLATPAK_DEST}/bin/tuxguitar"
+      - install -Dm0644 "${FLATPAK_DEST}/share/applications/tuxguitar.desktop" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - desktop-file-edit --set-key=Exec --set-value='tuxguitar %F' "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - desktop-file-edit --set-key=Icon --set-value="${FLATPAK_ID}" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
```
Comment (hfiguiere) [inline ar.com.tuxguitar.TuxGuitar.yml:57]:
```suggestion
      - desktop-file-edit --set-icon="${FLATPAK_ID}" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
```

**Comment by hfiguiere** [comment]
bot, build

**Comment by guiv42** [comment]
Thanks !

**Pull request merged**